### PR TITLE
Hippie Wizard Rework: General changes

### PR DESCRIPTION
**Pull request by Manly-Dwarf**
[Changelogs]: #

:cl: KawaiiBigBoss and McBawbaggings
add: Rebalanced Wizard.
balance: Wizard must now hold a staff (or mjolnir) in one of his hands to cast certain spells.
balance: Wizard now spawns with Magic Missile and Ethereal Jaunt, for free.
del: Lichwiz is fucking dead. May it never rest in peace.
tweak: Arcane Barrage and Lesser Summon Guns (the spell) have had their prices reduced to 2.
tweak: Plasma Fist has had its price upped to 3.
tweak: Eruption has had its price reduced to 1. Should honestly be free.
tweak: Lightning Bolt has had its price reduced to 2. Can only be cast with a staff.
tweak: Tesla Blast has had its price reduced to 1.
tweak: Mjolnir and the Singularity Hammer have had their prices reduced to 1.
tweak: Numerous spells now require a Staff to use. They are as follows: Magic Missile, Tesla Blast, Rod Form, Blink, Time Stop, Spacetime/Phase Distortion, Teleport, Mutate, Lightning Bolt, Soul Flare, Cluwne Curse, Eruption.
/:cl:

[why]: #

# **Wizard is annoying to fight against.**
Anyone can tell you this. Having an overpowered antagonist who can run around critting people with utmost ease but also can escape from any dangerous situation is not fun for the receiving players. Fighting a wizard who is aggressive can actually be quite fun, it's just the times when the Wizard merely spams Time Stop or other spells when you want to tear your hair out.
So that's what this PR is set out to fix. Make Wizard less annoying by removing his ability to cheat death, but still allowing him to have fun/be aggressive.

## **BALANCE CHANGES:**

**Global:**
Wizard now requires a staff to cast certain spells.
_Why:_ This prevents Wizard from casting certain spells while stunned to get out of jail for free. If the Wizard gets stunned, he's just as toast as anyone else. After we nerfed stuns to require upwards of two shots, the Wizard is a pain in the neck to fight as the moment you land a shot he will spam Magic Missile, Time Stop, Phase Distortion, etc. to keep you away.
Now Wizards who charge in and get shot can actually be killed. Wizard has to play like a smart person instead of not caring about anything, and only dying when he fireballs himself.

Wizard now has Magic Missile and Ethereal Jaunt for free.
_Why:_ Every Wizard buys these. Every single one. If you say that you don't, congratulations you're an almost extinct species who we now need to throw into a cage so we can repopulate your kind. Due to the Wizard taking a hit to his "I can escape death" side, it makes sense to throw him a bone and not force him to waste 4 points on two spells that everyone gets.
Plus, Magic Missile got nerfed.

**Fireball:**
Can no longer be cast while crit or handcuffed.
_Why:_ The war against escaping death. If I were to make it so that Wizards couldn't Stop Time any more, they would all just start spamming this in order to ensure they can never get stunned. You can still cast it without the need of a staff, but if you get stunned or handcuffed then you can't cast it.

**Plasma Fist:**
Price increased to 3 (from 1).
_Why:_ With the advent of "free magic missile" and the fact that you can just run up to a crowd, use any stun ability, and start gibbing people, this was a necessity. This is the same price as Magic Missile + Plasma Fist right now as it stands - so too much shouldn't change.
This may see some changes in a future PR.

**Mjolnir and Singularity Hammer:**
Price reduced to 1 (from 2).
_Why:_ Both items are two-handed, can't fit in backpacks, and force the Wizard to get up in people's faces and go to town. Now that the Wizard has to take a much bigger risk when he charges at people, this was something of a kindness.
You can also cast spells if you have one of these in hand. No need to worry about carrying a staff and a hammer.

**Arcane Barrage / Lesser Summon Guns:**
Price reduced to 2 (from 3).
_Why:_ They're both about as powerful as a Laser Gun. With the recent changes you can't even use it to its fullest potential unless you drop your staff and start firing away. Not worth 3 points.

**Soul Link/RISE/Lichwiz:**
Fucking dead.
_Why:_ A poorly thought out spell that shouldn't have existed to begin with. Wizard can already be a pain in the ass to kill, but giving him the ability to indefinitely come back to life with zero cost is unbelievable. That, and it made you spaceproof, syringeproof, and fireproof for unknown reasons.
What's the downsides? None at all. What a fun spell!

**Tesla Blast**
Price reduced to 1 (from 2).
_Why:_ Dude, this spell sucks. A charging time, a decently long cooldown, and minimal damage. /tg/ hasn't even touched this for over a year, and I doubt it gets any attention even on that side.

**Lightning Bolt**
Price reduced to 2 (from 3).
_Why:_ It's a strong spell, but Fireball can do the same-but-better and for a significantly shorter cooldown. The one benefit it has is being quick and hitting multiple people.

**Staves:**
Weight reduced from BULKY to NORMAL.
_Why:_ It's inevitable that a Wizard is going to get disarmed and lose their staff. If this happens and they escape alive, they're now worthless unless they purchased Recall. This helps alleviate the need for the Recall spell by letting the Wizard pack up extra staves in his backpack.

**Staff Requirements**
The following spells now require a staff to use:
Magic Missile, Tesla Blast, Rod Form, Blink, Time Stop, Spacetime/Phase Distortion, Teleport, Mutate, Lightning Bolt, Soul Flare, Cluwne Curse, Eruption.
_Why:_ A lot of these spells are Wizard's mobility spells, or powerful spells that can be cast while stunned. Having a Wizard run around with a shield and a laser gun, spamming Time Stop / Magic Missile and then instantly blasting everyone is now a tid harder, and the Wizard can now be crippled by taking away his favorite toy.
This also actually encourages people to buy staves.

## To be added:
NOTHING, ALL THAT SHIT GOES INTO ANOTHER PR.

Thanks to Bawbaggings for the help with this, my git had a nuclear meltdown.

### CODE REVIEW

**Comment by Manly-Dwarf** [comment]
Why am I conflicting with my own pull requests that were merged a while ago?
Github fucking sucks.

**Comment by Manly-Dwarf** [comment]
they went away

**Comment by Pyko1** [comment]
I guess this could play pout nicely.
I'm all in for the nerfs to wizards "get out of jail" cards.
And the free magic missile and ethreal jaunt is finally going to bring at least some change to wizard spell arsenals.

+1

**Comment by Pyko1** [comment]
Maybe a bit more discussion before making any decisions.

**Comment by Manly-Dwarf** [comment]
Add the controversial tag, because there's gonna be that one dude raging in the comments.

**Comment by JohnMcLaime** [comment]
Dont forget about shapeshift, it is basically anti stun and can be used robeless.
Fist is already better than plasma fist, can be used robeless too, needs only 1 hit to fuck someone over, stays upon mind transfer.

This change will make killing wizard too easy, maybe add multiple wizards then? Or make wiz drop something upon death which will turn whoever uses it into another wizard. Otherwise the gamemode will still have its problems of single antag and it turning into extended on their death basically,

**Comment by ghost** [comment]
I think making staves normal sized isn't the best change.
Perhaps add a wizard spell on a longish CD you start out with that conjured a new one? Encouraging players to just carry more staffs in their backpack or be unable to cast spells isn't great either.

**Comment by KayeArray** [comment]
>liking your own post

**Comment by praisenarsie** [comment]
> balance: Wizard now spawns with Magic Missile and Ethereal Jaunt, for free.

local man is stupid

**Comment by Manly-Dwarf** [comment]
@JohnMcLaime Doesn't the round end upon the Wizard's death now? Only if the shuttle can be called will the round not end when he dies. I could be wrong.
I was going to add more spells and get more into specifics, but I got told to mash things up into multiple PRs. Ultimately it's gotta be this way any how.

@McBawbaggings The point is to make Wizard not charge in like an idiot. If you can just instantly conjure a new one, then that defeats that purpose. Anyone who just rushes in besides KA-NADA usually doesn't have a long life, doesn't matter who you are. It's like if a Traitor rushes in and loses his ebow, HoS rushes in and loses his special gun, Cult loses their tome/sword, etc. Making staves normal sized won't change much.
Wizard right now can just charge in, get kills extremely easily, and then spam certain spells to keep himself from dying when he inevitably gets fucked over. If you're worried about him dying _too_ easily, he can still use EMP, Repulse, Blind, or other spells to mess with people attacking him while he's down. Just not straight up killing them or stunning them or just making himself invincible. Wizard should still be able to murderbone easily, it's just the staying alive part that's maybe a tid too easy.

@praisenarsie Local man has no argument, gets mad over inconsequential feature.

If you guys have any suggestions for the PR, I'd like to see them coming. I'll do individual spell changes next and then spell additios.

**Comment by ghost** [comment]
I just think carrying a bunch of staffs is as engaging gameplay as carrying 4 laser guns because it's easier to cycle them than just recharge them and such.

**Comment by DaydreamPuppeh** [comment]
I think that blink should not require a staff, but also cannot be used while stunned.

**Comment by Manly-Dwarf** [comment]
Added a list of things I'm currently working on, just in case anyone needs to know.

**Comment by Carbonhell** [comment]
1)I don't like the w_class change. Should keep them bulky,so they'll be more careful to not lose their staff.
2)Blink is robeless, it should not require the staff aswell.
The rest seems ok, it could be testmerged yea, +1

File: code/modules/spells/spell.dm [hunk 1]
```diff
@@ -208,6 +211,15 @@ GLOBAL_LIST_INIT(spells, typesof(/obj/effect/proc_holder/spell)) //needed for th
 			if(!is_type_in_typecache(H.head, casting_clothes))
 				to_chat(H, "<span class='notice'>I don't feel strong enough without my hat.</span>")
 				return 0
+		/* hippie edit */
+		if(staff_req)
```
Comment (Carbonhell) [inline code/modules/spells/spell.dm]: 1) Why not make staff_req a typecached list so you can do everything with one simple check?
2) I think you can modularize this by making a new file and doing
/obj/effect/proc_holder/spell/cast_check(skipcharge = 0,mob/user = usr)
..()
your stuff

Comment (Manly-Dwarf) [reply code/modules/spells/spell.dm]: Typecache would be a pain and doesn't check the whole list of items (if I recall), so every item that I wanted to be compatible with magic I would have to write down, and out of fear for missing items I just used ifs.

I'll try out the modularized version soon. If it doesn't work I'll report back.

Comment (tortellinitony) [reply code/modules/spells/spell.dm]: make a datum component :)

**Comment by tortellinitony** [comment]
plasma fist does not need its price upped

**Comment by Manly-Dwarf** [comment]
Yes it does. It has been forgotten with the changes within the community, but Plasma Fist being hilariously cheap and functioning as an upgraded Ei Nath with no cooldown is disgusting.
I abused the shit out of it when I rolled Wizard way back when. It hasn't been touched at all since then, it works the exact same.

**Comment by tortellinitony** [comment]
you might as well remove it then, if nobody uses it now then actually nobody will use it at a cost of 3

**Comment by 2Dollarbill** [comment]
I am a bit mixed on a few changes, however I think it's worth giving a try. The plasmafist change is very much needed as much as people like to say it's pointless since nobody uses it. We still have explosive lances the way they are. +1.

**Comment by Manly-Dwarf** [comment]
This has +2, so I'll hurry up and finish it.

**Comment by Manly-Dwarf** [comment]
(Bawb) I may have nuked my git on accident, so I can't actually merge any changes I've added in the future.
I'll add the spells (including Conjure Staff) in a separate PR, which is honestly how we were supposed to do it to begin with.
tl;dr merge dis

**Review by praisenarsie — DISMISSED** [review]

File: code/modules/spells/spell.dm (hunk 1 above)
Comment (ghost) [reply code/modules/spells/spell.dm]: please do what @tortellinitony said

**Comment by Manly-Dwarf** [comment]
All right, I have locked the jew in my dungeon and he will fix the meltdown he caused.
Should be able to do the component soon.

**Comment by Manly-Dwarf** [comment]
The jew has fixed Chernobyl.
@Chnkr @tortellinitony everything's now a datum component

**Comment by praisenarsie** [comment]
fuck i wished this could be testmerged...

**Comment by ghost** [comment]
minor bug:
it should be anything with a catalyst gets a special examine text telling you it has a magical glow
i didn't get it working but it correctly acts as a catalyst and spells correctly check for it

**Comment by Manly-Dwarf** [comment]
Bug is fixed.

**Comment by tortellinitony** [comment]
-1

plasma fist doesn't deserve a price increase if you aren't going to make any improvements to it

**Comment by Manly-Dwarf** [comment]
Fuck off, Tony. Plasma Fist is quite literally an infinite, no-recharge Ei Nath with several other enhancements. Just because people don't use it doesn't mean it isn't hilariously broken (e.g. Explosive Lance).

**Comment by Pyko1** [comment]
+1 for change, if there are issues we can also spiraldown like tg in endless rebalancing now!

**Comment by Pyko1** [comment]
welp needs to be reviewed, absolute autism.

**Comment by ghost** [comment]
Your Fix/Feature pull request delta is now above zero (0). Feel free to make Feature/Tweak/Balance PRs.
